lib/ansible/executor/task_queue_manager.py

```diff
@@ -187,7 +187,8 @@ def run(self, play):
 
         new_play = play.copy()
         new_play.post_validate(templar)
-        new_play.handlers = new_play.compile_roles_handlers() + new_play.handlers
+        for handler in new_play.compile_roles_handlers():
+            new_play.add_handler(handler)
```

Comment:
Isn't this changing handlers order? Role handlers are currently prepended to the play handlers. This appends them, no?

Reply:
I think this line is actually useless: `compile_roles_handlers()` calls `get_handler_blocks()` [1], and `get_handler_blockers()` already refresh the `handlers` list [2].

So `new_play.add_handler(handler)` here loops on a list of handlers that are already in new_play.handlers.

[1]: https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/play.py#L250
[2]: https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role/__init__.py#L385